File: cis_interface/examples/gs_lesson4/src/gs_lesson4_modelA.m
% Initialize input/output channels 
in_channel = PsiInterface('PsiInput', 'inputA');
out_channel = PsiInterface('PsiOutput', 'outputA');

% Receive input from input channel
% If there is an error, the flag will be False.
res = in_channel.recv();
flag = res{1};
msg = res{2};

% Print received message
fprintf('%s\n', char(msg));

% Send output to output channel
% If there is an error, the flag will be False
flag = out_channel.send(msg);

exit(0);

File: cis_interface/examples/gs_lesson4/src/gs_lesson4_modelA.py
# Import classes for input/output channels
from cis_interface.interface.PsiInterface import PsiInput, PsiOutput

# Initialize input/output channels
in_channel = PsiInput('inputA')
out_channel = PsiOutput('outputA')

# Loop until there is no longer input or the queues are closed
while True:

    # Receive input from input channel
    # If there is an error, the flag will be False
    flag, msg = in_channel.recv()
    if not flag:
        print("Model A: No more input.")
        break

    # Print received message
    print('Model A: %s' % msg)

    # Send output to output channel
    # If there is an error, the flag will be False
    flag = out_channel.send(msg)
    if not flag:
        print("Model A: Error sending output.")
        break
